### PR TITLE
Refactor out unique assertion

### PR DESCRIPTION
**Pull request by mattsaxon**
As per code review comments from https://github.com/jstasiak/python-zeroconf/pull/217

### CODE REVIEW

File: zeroconf/__init__.py [hunk 1]
```diff
@@ -882,21 +882,15 @@ class State(enum.Enum):
...
-
-        """Only support for unique answers"""
-        if (
-            record.type == _TYPE_TXT
-            or record.type == _TYPE_SRV
-            or record.type == _TYPE_A
-            or record.type == _TYPE_AAAA
-        ):
-            assert record.unique
-
```
Comment (jstasiak) [inline zeroconf/__init__.py]: Do we not want this anymore?

File: zeroconf/__init__.py [hunk 2]
```diff
@@ -905,13 +899,7 @@ def add_answer_at_time(self, record: Optional[DNSRecord], now: Union[float, int]
...
 
-            """Only support for unique answers"""
-            if (
-                record.type == _TYPE_TXT
-                or record.type == _TYPE_SRV
-                or record.type == _TYPE_A
-                or record.type == _TYPE_AAAA
-            ):
+            if self.is_type_unique(record.type):
                 assert record.unique
```
Comment (jstasiak) [inline zeroconf/__init__.py]: We could put the assert in the function/method too and call it `assert_unique_used_correctly` or something I think.

File: zeroconf/__init__.py (hunk 1 above)
Comment (mattsaxon) [reply zeroconf/__init__.py]: I noticed when refactoring that it wasn't needed as the same assertion is called inside the called function add_answer_at_time()

File: zeroconf/__init__.py (hunk 2 above)
Comment (mattsaxon) [reply zeroconf/__init__.py]: I considered it, but thought it was more readable as I have it. I pondered it for a while and decided the readability was better than saving the one line. I'd appreciate your view to be honest and could have gone either way.

File: zeroconf/__init__.py [hunk 3]
```diff
@@ -882,21 +882,15 @@ class State(enum.Enum):
         init = 0
         finished = 1
 
+    @staticmethod
+    def is_type_unique(type_: int) -> bool:
+        return type_ == _TYPE_TXT or type_ == _TYPE_SRV or type_ == _TYPE_A or type_ == _TYPE_AAAA
```
Comment (jstasiak) [inline zeroconf/__init__.py]: Maybe just `type_ in {_TYPE_TXT, _TYPE_SRV, _TYPE_A, _TYPE_AAAA}`?

File: zeroconf/__init__.py (hunk 2 above)
Comment (jstasiak) [reply zeroconf/__init__.py]: That's fair, looks good either way.

**Comment by jstasiak** [comment]
Sorry, I forgot about this pull request – since you pushed the commit as part of #228 I merged it from there and this can be closed.

**Pull request closed without merging**